src/main/java/com/extrahardmode/features/AnimalCrowdControl.java

```diff
@@ -0,0 +1,173 @@
+            int density = 0;
+
+          /**
+            *Loop and check if entity is an animal while 
+            *looping count how many animals have spawned
+            * by incrementing density
+            */
+            for (Entity a : cattle) {
+                if (a instanceof Animals
+                        && a.getType() != EntityType.HORSE
+                        && a.getType() != EntityType.WOLF
+                        && a.getType() != EntityType.OCELOT) {
+                    density++;
+                    
+                    //Check if the amount of animals is bigger than the threshold given
+                    if (density > threshold) {
+                        final LivingEntity animal = (LivingEntity) a;
+                        
+                      /**
+                        *This creates a runnable assign to each animals will close once
+                        *if animal is far enough from other animals or animal is dead
+                        */ 
+                        new BukkitRunnable() {
+
+                            boolean firstRun = true;
+
+                            @Override
+                            public void run() {
+                                List<Entity> cattle = e.getNearbyEntities(3, 3, 3);
+                                int density = 0;
+
+                                //this will be used to check if animal is far from other animals
+                                for (Entity a : cattle) {
+                                    if (a instanceof Animals
+                                            && a.getType() != EntityType.HORSE
+                                            && a.getType() != EntityType.WOLF
+                                            && a.getType() != EntityType.OCELOT) {
+                                        density++;
+                                    }
+                                }
+                                
+                                if (animal.isDead() || density <= threshold) {
+                                    this.cancel();
+                                } else {
+                                    /**
+                                     *Hack to force animal to move away exploits the default AI of animals
+                                     *the set Velocity make sure that no knockback is given
+                                     */
+                                    if (firstRun) {
+                                        firstRun = false;
+                                        animal.damage(0, animal);
+                                        animal.setVelocity(new Vector());
+                                    } else {
+                                        animal.damage(0.5);
+                                    }
+                                }
+                            }
+                        }.runTaskTimer(this.plugin, 20, 20);

```

Comment:
As I have tested for awhile now aside from the effect of seeing your animals getting damage the damage given does give the animal some time to spread out in most of the situation I have been thinking of just merging animal.damage(0,animal) and animal.damage(0.5) along with setVelocity you might notice why I have two different damage calls it has something to do with the checking to remove animal drops. But then 5-10 second delay, I am also alright.